### PR TITLE
feat(docs): new theme

### PR DESCRIPTION
**Pull request by userquin**
Pending `Vitepress` isssues/PR:
- https://github.com/vuejs/vitepress/issues/758 (we can publish docs without this)
- https://github.com/vuejs/vitepress/issues/760 (fixed via plugin: required to make it working, check https://github.com/vuejs/vitepress/issues/795)
- https://github.com/vuejs/vitepress/pull/709 (we don't need this to publish docs, we're using `esno 0.14.1`)
- https://github.com/vuejs/vitepress/pull/746 (this is solved here manually)

### CODE REVIEW

File: docs/.vitepress/config.ts [hunk 1]
```diff
@@ -158,45 +175,30 @@ const Workbox = [
 const slidebars = [
   {
     text: 'Guide',
-    children: Guide.map((e) => {
-      (e as any).useLinkText = `${e.text} | Guide`
```
Comment (userquin) [inline docs/.vitepress/config.ts:162]: if https://github.com/vuejs/vitepress/issues/758 is included we can use it here

Comment (userquin) [reply docs/.vitepress/config.ts:162]: maybe we can send a PR

**Comment by userquin** [comment]
We also need to fix the bg-color for the code block inside warnings: https://deploy-preview-304--vite-plugin-pwa.netlify.app/guide/development.html#injectmanifest-strategy

**Review by antfu — APPROVED** [review]

**Comment by userquin** [comment]
changing docs and testing https://github.com/vuejs/vitepress/issues/809

**Pull request merged**